modules/Multimode-Networks/src/main/java/cz/cvut/fit/gephi/multimode/LongTaskTransformation.java

```diff
@@ -0,0 +1,193 @@
+            return;
+        }
+        Progress.progress(progressTicket, "Removing nodes/edges");
+        
+        
+        if (removeNodes) {
+            for (Node n : firstHorizontal) {
+                graph.removeNode(n);
+            }
+        } else {
+            if (removeEdges) {
+                for (int i = 0; i < firstMatrix.getM(); i++) {
+                    for (int j = 0; j < firstMatrix.getN(); j++) {
+                        if (graph.contains(firstVertical.get(i)) && graph.contains(firstHorizontal.get(j)) && graph.getEdge(firstVertical.get(i), firstHorizontal.get(j)) != null && firstMatrix.get(i, j) > 0) {
+                            graph.removeEdge(graph.getEdge(firstVertical.get(i), firstHorizontal.get(j)));
+                        }
+                    }
+                }
+                
+                for (int i = 0; i < secondMatrix.getM(); i++) {
+                    for (int j = 0; j < secondMatrix.getN(); j++) {
+                        if (graph.contains(secondVertical.get(i)) && graph.contains(secondHorizontal.get(j)) && graph.getEdge(secondVertical.get(i), secondHorizontal.get(j)) != null && secondMatrix.get(i, j) > 0) {
+                            graph.removeEdge(graph.getEdge(secondVertical.get(i), secondHorizontal.get(j)));
+                        }
+                    }
+                }
+            }
+        }
+        
+        if (cancelled) {
+            return;
+        }
+        Progress.progress(progressTicket, "Creating new edges");
+        /*AttributeController ac = Lookup.getDefault().lookup(AttributeController.class);
+        AttributeModel model = ac.getModel();        
+        AttributeColumn edgeTypeCol = model.getEdgeTable().getColumn("MMNT-EdgeType");        
+        if (edgeTypeCol == null) {
+            edgeTypeCol = model.getEdgeTable().addColumn("MMNT-EdgeType", AttributeType.STRING);
+        }*/
+        
+        Table edgeTable = graphController.getGraphModel().getEdgeTable();
+        Column MMNT;
+        if(!edgeTable.hasColumn("MMNT-EdgeType")){
+            MMNT = edgeTable.addColumn("MMNT-EdgeType", String.class);
+        } else {
+            MMNT = edgeTable.getColumn("MMNT-EdgeType");
+        }
+        
+        for (int i = 0; i < result.getM(); i++) {
+            for (int j = 0; j < result.getN(); j++) {
+                if (graph.contains(firstVertical.get(i)) && graph.contains(secondHorizontal.get(j)) && graph.getEdge(firstVertical.get(i), secondHorizontal.get(j)) == null && result.get(i, j) > 0) {
+                    Edge ee = graphModel.factory().newEdge(firstVertical.get(i), secondHorizontal.get(j),1, (float) result.get(i, j), graphModel.isDirected());
+                    if (!ee.isSelfLoop()) {
+                        
+                        ee.setAttribute(MMNT, inDimension + "<--->" + outDimension);
+                        ee.setLabel(inDimension + "-" + outDimension);
+                        graph.addEdge(ee);
```

Comment:
This part of the code should check if the same edge already exists before adding it.

I uploaded sample code for this and other changes here: https://github.com/eduramiba/gephi-plugins/tree/multimode-network